extensions/csv/src/test/java/io/deephaven/csv/CsvTest.java

```diff
@@ -1,529 +1,638 @@
-                .newTable();
-        return new CsvTest("timestampNanos", "timestamp-nanos.csv", CsvSpecs.csv(), expected);
-    }
-
-    public static CsvTest timestampMixed() {
-        // Can't infer milli and micros in a single column - will parse as a long, not an Instant.
-        final NewTable expected = ColumnHeader.ofLong("Timestamp")
-                .row(TIMESTAMP_A.toEpochMilli())
-                .row(null)
-                .row(TIMESTAMP_B.toEpochMilli() * 1000L)
-                .newTable();
-        return new CsvTest("timestampMixed", "timestamp-mixed.csv", CsvSpecs.csv(), expected);
-    }
-
-    public static CsvTest timestampLegacy() {
-        final NewTable expected = ColumnHeader.ofInstant("Timestamp")
-                .row(TIMESTAMP_A)
-                .row(null)
-                .row(TIMESTAMP_B)
-                .newTable();
-        return new CsvTest("timestampLegacy", "timestamp-legacy.csv", CsvSpecs.csv(), expected);
-    }
-
-    public static CsvTest bools() {
-        final NewTable expected = ColumnHeader.ofBoolean("Bool")
-                .row(true)
-                .row(null)
-                .row(false)
-                .row(true)
-                .row(false)
-                .row(true)
-                .row(false)
-                .newTable();
-        return new CsvTest("bools", "bools.csv", CsvSpecs.csv(), expected);
-    }
-
-    public static CsvTest chars() {
-        final NewTable expected = ColumnHeader.ofChar("Char")
-                .row('A')
-                .row(null)
-                .row('B')
-                .row('C')
-                .row('1')
-                .row('2')
-                .row('3')
-                .newTable();
-        return new CsvTest("chars", "chars.csv", CsvSpecs.csv(), expected);
-    }
-
-    public static CsvTest byteViaHeader() {
-        final NewTable expected = ColumnHeader.ofByte("Byte")
-                .row((byte) (Byte.MIN_VALUE + 1))
-                .row(null)
-                .row(Byte.MAX_VALUE)
-                .newTable();
-        return new CsvTest("byteViaHeader", "byte.csv", CsvSpecs.builder().header(expected.header()).build(), expected);
-    }
-
-    public static CsvTest byteViaInference() {
-        final NewTable expected = ColumnHeader.ofByte("Byte")
-                .row((byte) (Byte.MIN_VALUE + 1))
-                .row(null)
-                .row(Byte.MAX_VALUE)
-                .newTable();
-        return new CsvTest("byteViaInference", "byte.csv",
-                CsvSpecs.builder().inference(InferenceSpecs.builder().addParsers(Parser.BYTE).build()).build(),
+    @Test
+    public void timestampMillis() {
+        final String input = "" +
+                "Timestamp\n" +
+                "1632769200000\n" +
+                "\n" +
+                "1632772800000\n";
+
+        final Table expected = TableTools.newTable(
+                TableTools.col("Timestamp", DATETIME_A, null, DATETIME_B)
+        );
```

Comment:
If you want to rely on the existing tests, you can easily translate from NewTable to engine Table: `io.deephaven.engine.table.impl.InMemoryTable#from`

Reply:
I think I went with engine Table because I wanted to make sure ColumnSources worked, I wanted to do a large null-heavy (i.e. sparse) test, and also QST table didn't yet have the support for datetime-as-long that I thought was important, so I cut over to engine Table.